api/java/manipulating-databases/db_list.md

```diff
@@ -22,5 +22,9 @@ List all database names in the cluster. The result is a list of strings.
 __Example:__ List all databases.
 
 ```java
-r.dbList().run(conn);
+List<?> dbList = r.dbList().run(connection, ArrayList.class).single();
+
+if (dbList != null) {
```

Comment:
I'm not familiar with the Java driver, but I think dbList will never be null.  Unless RethinkDB returns a nil result.

Reply:
In this example, my IDE (IntelliJ IDEA) complained that `forEach` could produce a `NullPointerException` and suggested adding this check.

In the end I did something more like this:

```java
final Result<?> result = r.dbList().run(connection, List.class);

try (result) {
    List<?> databaseList = (List<?>) result.single();

    if (databaseList != null) {
        databaseList.forEach(System.out::println);
    }
}
```

So, maybe that should be the example code? People can still leave out anything they do not want or rewrite it according to their needs. My thought was just to provide a hint on how to use it. And maybe a more elaborated example could also lower the hurdle for new developers, to see how they could use it.

But of course the provided example should show a somewhat recommended way. For me `r.dbList().run(conn);` was not really that sufficient or telling and this is what I thought could have helped me in the first place to get to use it more quickly. This is code I use because it is robust, but any suggestions are welcome.. I am referring to the new code above.